osu.Game/Screens/Select/Carousel/CarouselBeatmap.cs

```diff
@@ -59,6 +59,8 @@ private bool checkMatch(FilterCriteria criteria)
                      criteria.Artist.Matches(BeatmapInfo.Metadata.ArtistUnicode);
             match &= !criteria.Title.HasFilter || criteria.Title.Matches(BeatmapInfo.Metadata.Title) ||
                      criteria.Title.Matches(BeatmapInfo.Metadata.TitleUnicode);
+            match &= !criteria.DifficultyName.HasFilter || criteria.DifficultyName.Matches(BeatmapInfo.DifficultyName) ||
+                     criteria.DifficultyName.Matches(BeatmapInfo.Metadata.TitleUnicode);
```

Comment:
I'm not sure why this checks `TitleUnicode`?

It actually partially handles the case I mentioned above, but if that was the intent, then why isn't it checking `Title` too?

Or, it shouldn't be checking either, it should only check `DifficultyName`; a workaround that will work for searching beatmaps with brackets in the track title is using double quotes, i.e. `"[game edit]"`.